packages/client/src/components/Actions/ActionsListTable.js

```diff
@@ -0,0 +1,137 @@
+import { useContext } from "react";
+import { useErrorMessage } from "hooks";
+import { formatActionString } from "utils";
+import { useModalContext } from "contexts";
+import { Web3Context } from "contexts/Web3";
+import { SIGNER_RESPONSES } from "constants/enums";
+import Svg from "library/Svg";
+import ApprovalRequired from "./components/ApprovalRequired";
+
+const Row = ({
+  safeData,
+  action,
+  displayIndex,
+  onApprove,
+  onReject,
+  onExecute,
+}) => {
+  const onApproveAction = async (action) => {
+    try {
+      await onApprove(action);
+      closeModal();
+    } catch (error) {
+      showErrorModal(error);
+    }
+  };
+
+  const onRejectAction = async (action) => {
+    try {
+      await onReject(action);
+      closeModal();
+    } catch (error) {
+      showErrorModal(error);
+    }
+  };
+
+  const openApproveOrRejectModal = async (action) => {
+    const view = await getActionView(safeData.address, action.uuid);
+    openModal(
+      <ApprovalRequired
+        safeData={safeData}
+        actionView={view}
+        confirmations={action.signerResponses}
+        onApprove={() => onApproveAction(action)}
+        onReject={() => onRejectAction(action)}
+      />,
+      { headerTitle: "Approval Required" }
+    );
+  };
+  const totalSigned = Object.values(action.signerResponses).filter(
+    (x) => x === SIGNER_RESPONSES.APPROVED
+  ).length;
+  const executionReady = totalSigned >= safeData.threshold;
+  const background = executionReady ? "has-text-danger" : "has-text-warning";
+  const actionPrompt = executionReady
+    ? "Pending Signature"
+    : "Pending Approval";
+  const actionText = executionReady ? "Sign" : "Approve";
+  const actionSvg = executionReady ? "Quill" : "EmptyCheck";
+  const actionFn = executionReady ? onExecute : openApproveOrRejectModal;
+
+  const { getActionView } = useContext(Web3Context);
+  const { showErrorModal } = useErrorMessage();
+  const { openModal, closeModal } = useModalContext();
+
+  return (
+    <>
+      <tr className="py-4 is-flex is-align-items-center">
+        <td className="p-3 flex-1">{String(displayIndex).padStart(2, "0")}</td>
+        <td className="p-3 flex-7">{formatActionString(action.intent)}</td>
+        <td className="p-3 flex-4 is-hidden-touch">
+          <Svg key={background} name="Status" className={background} />
```

Comment:
what does the key property do?

Reply:
There was an issue - color didn't change for the svg once signer approved the action, so when key is changed the color is updated as well

Reply:
Got it, I'll fix the Svg loader